lib/sles4sap_publiccloud.pm

```diff
@@ -667,4 +673,96 @@ sub create_hana_vars_section {
     return (\%hana_vars);
 }
 
+=head2 get_hana_database_info
+
+    Parses "hdbsql" command output.
+    Returns 1 if hana database is online otherwise 0 offline.
+
+=back
+=cut
+
+sub get_hana_database_info {
+    my ($self, $password_db, $instance_id) = @_;
+    my $hana_database_online = '"HDB","HDB-$instance_id","YES","ACTIVE","","","DEFAULT","?"';
+
+    sleep 30;
+    eval {
+        my $output_cmd = $self->run_cmd(cmd => "hdbsql -u SYSTEM -p $password_db -i $instance_id 'SELECT * FROM SYS.M_DATABASES;'", runas => get_required_var("SAP_SIDADM"));
+    };
+    if ($@) {
+        record_info('HANA DB OFFLINE', "Hana database in primary node is offline. Here the error \n$@");
+        return 0;
+    }
+    else {
+        return 1;
+    }
+}
+
+=head2 is_hana_database_offline
+
+    Check if hana DB is offline.
+    Returns if hana database is online with 1 and offline with 0
+
+=over 1
+
+=item B<TIMEOUT> - default 900
```

Comment:
`total_consecutive_passes` argument documentation is missing

Reply:
thx